git/revparse.go

```diff
@@ -270,7 +272,67 @@ func RevParseCommitish(c *Client, opt *RevParseOptions, arg string) (cmt Commiti
 	if b, err := GetBranch(c, cmtbase); err == nil {
 		return b, nil
 	}
-	return nil, fmt.Errorf("Could not parse %v", arg)
+
+	// Try seeing if it's an abbreviation of a commit as a last
+	// resort. We require a length of at least 3, so that we only
+	// need to search one directory of the objects directory.
+	if len(cmtbase) > 2 && len(cmtbase) < 40 {
+		dir := cmtbase[:2]
+		var candidates []CommitID
+
+		files, err := ioutil.ReadDir(c.GitDir.File(File("/objects/" + dir)).String())
```

Comment:
I wonder if we should be trying to use filepath.Join() more often when composing paths.

Reply:
sure, I'll fix it tonight when I'm home from work.

Reply:
Done.. as a bonus, it also looks cleaner to read